tests/cache/test-vary-cache.php

```diff
@@ -580,4 +580,121 @@ public function test__remove_nocache_for_user() {
 
 		$this->assertEquals( 1, did_action( 'vip_vary_cache_did_send_headers' ) );
 	}
+
+
+
+	public function get_test_data__stringify_groups() {
+		return [
+			'values_for_all_groups' => [
+				[
+					'dev-group',
+					'design-group',
+				],
+				[
+					'dev-group' => 'yes',
+					'design-group' => 'no',
+				],
+				'vc-v1__design-group_--_no---__dev-group_--_yes'
+			],
+			'values_for_only_nonempty_groups' => [
+				[
+					'dev-group',
+					'design-group'
+				],
+				[
+					'dev-group' => 'yes',
+				],
+				'vc-v1__dev-group_--_yes'
+			],
+			'values_for_all_empty_groups' => [
+				[
+				],
+				[
+				],
+				'vc-v1__'
```

Comment:
If the groups are empty, should we just return empty as well? And then `set_group_cookie` would need to either `unset_cookie` or just skip setting the cookie.

Reply:
Perhaps. Although don't we need to set the header and cookie in any case?

Reply:
The Vary header will get set thanks to `register_groups`.

The downside to this mostly empty cookie is that we end up with two groups with the same content (users with no cookie, and users with mostly empty cookie) which reduces our cache efficacy.

Reply:
Okay, I've addressed this. Setting the string to null in the case of no groups, which I _think_ is right.